packages/sdk-react/src/components/counterparts/CounterpartDetails/CounterpartAddressFormUpdate/useCounterpartAddressFormUpdate.tsx

```diff
@@ -56,14 +54,23 @@ export function useCounterpartAddressFormUpdate({
   }, [formRef]);
 
   const updateAddress = useCallback(
-    (payload: CounterpartUpdateAddress) => {
+    (payload: components['schemas']['CounterpartUpdateAddress']) => {
       if (!address) return;
 
-      return addressUpdateMutation.mutate(payload, {
-        onSuccess: () => {
-          onUpdate && onUpdate(address[0].id);
+      return addressUpdateMutation.mutate(
+        {
+          path: {
+            counterpart_id: address.data[0].counterpart_id,
```

Comment:
The component is supposed to update the certain counterpart address by these two parameters from props: `addressId`, `counterpartId`.
We can't take these parameters from `data[0]` (There is no guarantee we have the address we want to update or the desired list order).
So, pass the props `addressId`, `counterpartId`